src/TestLogger/Core/TestRunResultWorkflow.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Spekt Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Spekt.TestLogger.Core
+{
+    using System;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+    public static class TestRunResultWorkflow
+    {
+        public static void Result(this ITestRun testRun, TestResultEventArgs resultEvent)
+        {
+            testRun.Store.Add(new TestResultInfo(resultEvent.Result));
+#if NONE
+            if (TryParseName(result.TestCase.FullyQualifiedName, out var typeName, out var methodName, out _))
```

Comment:
Do you think parsing is going to just sit as a static class, or will it added to the `TestRun`? In either case, are you planning to drop the constructor to `TestResultInfo` that requires the namespace, type and method? I think this is the logical point to do the name parsing, so the `ITestResultSeralizer` doesn't need to deal with it.

The Parser will need access to `IConsoleOutput` so it can write a debug message if the parsing ran into errors.

Reply:
To be honest, I had postponed this decision to a future PR. Not planning to drop the ctor for TestResultInfo yet. Question remains for us if the parser is encapsulated by the TestResultInfo (with the risk of mixing data and behavior). Let's iterate on this when we get to implementing the Result workflow.

Agree on the IConsoleOutput dependency to parser.